### ISSUE TITLE
rclone hanging after retries exceed for a file

### ISSUE THREAD

**Issue by calisro** [ncw/rclone#1217]
I searched other issues but I wasn't 100% sure if they were similar enough to bump them.  So here is a new one that can be tied together.  

I have two mounts.  1 GDrive and 1 ACD.  I push the same content with rclone itselv to both via mounts.  I get lots of errors on both.  The errors are more plentiful than the successes. :)

So the path looks like this:

local volume-> rclone copy /data/Media1 /data/Media3  where Media1 is local content and Media3 is a gdrive crypt mount.
local volume-> rclone copy /data/Media1 /data/Media2  where Media1 is local content and Media2 is a ACD crypt mount.

This is an example of the log output from ACD:

```
2017/03/06 07:30:02 INFO  : Videos/Movies/Movie1/Movie1.nfo: Copied (new)
2017/03/06 07:30:03 INFO  : Videos/Series/Series2/Season.1/.actors/Jason_Watkins.jpg: Copied (new)
2017/03/06 07:30:07 NOTICE: Videos/Series/Series3/Season.2/Series3.S02E03.720p.HDTV.x264-AVS[PRiME]/Torrent Downloaded From 1337x.to.txt: Removing partially written file on error: close /data/Media2/Videos/Series/Series3/Season.2/Series3.S02E03.720p.HDTV.x264-AVS[PRiME]/Torrent Downloaded From 1337x.to.txt: input/output error
2017/03/06 07:30:07 ERROR : Videos/Series/Series3/Season.2/Series3.S02E03.720p.HDTV.x264-AVS[PRiME]/Torrent Downloaded From 1337x.to.txt: Failed to remove partially written file: remove /data/Media2/Videos/Series/Series3/Season.2/Series3.S02E03.720p.HDTV.x264-AVS[PRiME]/Torrent Downloaded From 1337x.to.txt: no such file or directory
2017/03/06 07:30:07 ERROR : Videos/Series/Series3/Season.2/Series3.S02E03.720p.HDTV.x264-AVS[PRiME]/Torrent Downloaded From 1337x.to.txt: Failed to copy: close /data/Media2/Videos/Series/Series3/Season.2/Series3.S02E03.720p.HDTV.x264-AVS[PRiME]/Torrent Downloaded From 1337x.to.txt: input/output error
2017/03/06 07:30:07 NOTICE: Videos/Series/Series4/Season.2/Series4.S02E10.720p.webrip.h264-thumb.jpg: Removing partially written file on error: close /data/Media2/Videos/Series/Series4/Season.2/Series4.S02E10.720p.webrip.h264-thumb.jpg: input/output error
2017/03/06 07:30:07 ERROR : Videos/Series/Series4/Season.2/Series4.S02E10.720p.webrip.h264-thumb.jpg: Failed to remove partially written file: remove /data/Media2/Videos/Series/Series4/Season.2/Series4.S02E10.720p.webrip.h264-thumb.jpg: no such file or directory
2017/03/06 07:30:07 ERROR : Videos/Series/Series4/Season.2/Series4.S02E10.720p.webrip.h264-thumb.jpg: Failed to copy: close /data/Media2/Videos/Series/Series4/Season.2/Series4.S02E10.720p.webrip.h264-thumb.jpg: input/output error
2017/03/06 07:30:08 NOTICE: Videos/Series/Series4/tvshow.nfo: Removing partially written file on error: close /data/Media2/Videos/Series/Series4/tvshow.nfo: input/output error
2017/03/06 07:30:08 ERROR : Videos/Series/Series4/tvshow.nfo: Failed to remove partially written file: remove /data/Media2/Videos/Series/Series4/tvshow.nfo: no such file or directory
2017/03/06 07:30:08 ERROR : Videos/Series/Series4/tvshow.nfo: Failed to copy: close /data/Media2/Videos/Series/Series4/tvshow.nfo: input/output error
2017/03/06 07:30:08 NOTICE: Videos/Series/Series4/Season.2/Series4.S02E03.webrip.h264-thumb.jpg: Removing partially written file on error: close /data/Media2/Videos/Series/Series4/Season.2/Series4.S02E03.webrip.h264-thumb.jpg: input/output error
2017/03/06 07:30:08 ERROR : Videos/Series/Series4/Season.2/Series4.S02E03.webrip.h264-thumb.jpg: Failed to remove partially written file: remove /data/Media2/Videos/Series/Series4/Season.2/Series4.S02E03.webrip.h264-thumb.jpg: no such file or directory
2017/03/06 07:30:08 ERROR : Videos/Series/Series4/Season.2/Series4.S02E03.webrip.h264-thumb.jpg: Failed to copy: close /data/Media2/Videos/Series/Series4/Season.2/Series4.S02E03.webrip.h264-thumb.jpg: input/output error
2017/03/06 07:32:51 INFO  : Videos/Series/Series5/Season.6/Series5.S06E04.720p.hdtv.h264.mkv: Copied (new)
2017/03/06 07:34:58 INFO  :
Transferred:   3.713 GBytes (12.675 MBytes/s)

```

These things retry and sometimes they are successful and sometimes they are not.  But overall there a LOT of errors/retries.  

This is hte log from the mount itself
```
2017/03/06 07:30:07 ERROR : ....... : WriteFileHandle.Flush error: HTTP code 429: "429 Too Many Requests": response body: "{\"logref\":\"9ea92930-0268-11e7-ad9f-43c113caf993\",\"message\":\"Rate exceeded\",\"code\":\"\"}"
2017/03/06 07:30:07 ERROR : ....... : Dir.Remove error: no such file or directory
2017/03/06 07:30:07 ERROR : ....... : WriteFileHandle.Release error: HTTP code 429: "429 Too Many Requests": response body: "{\"logref\":\"9ecd2bf3-0268-11e7-99a8-477cb4e7d4b9\",\"message\":\"Rate exceeded\",\"code\":\"\"}"
2017/03/06 07:30:07 ERROR : ....... _HIDDEN~: WriteFileHandle.Release error: HTTP code 429: "429 Too Many Requests": response body: "{\"logref\":\"9eedfaae-0268-11e7-a6cf-5d645af872ff\",\"message\":\"Rate exceeded\",\"code\":\"\"}"
2017/03/06 07:30:07 ERROR : ....... : WriteFileHandle.Flush error: HTTP code 429: "429 Too Many Requests": response body: "{\"logref\":\"9f0a0daa-0268-11e7-9179-fbbb7ecb6809\",\"message\":\"Rate exceeded\",\"code\":\"\"}"
2017/03/06 07:30:07 ERROR : ....... : Dir.Remove error: no such file or directory
2017/03/06 07:30:08 ERROR :....... : WriteFileHandle.Flush error: HTTP code 429: "429 Too Many Requests": response body: "{\"logref\":\"9f28b982-0268-11e7-ba9f-795965b51193\",\"message\":\"Rate exceeded\",\"code\":\"\"}"
2017/03/06 07:30:08 ERROR : ....... : Dir.Remove error: no such file or directory
2017/03/06 07:30:08 ERROR :....... : WriteFileHandle.Flush error: HTTP code 429: "429 Too Many Requests": response body: "{\"logref\":\"9f1690d3-0268-11e7-8b59-6db6d047a573\",\"message\":\"Rate exceeded\",\"code\":\"\"}"
2017/03/06 07:30:08 ERROR : ....... : Dir.Remove error: no such file or directory
2017/03/06 07:38:24 ERROR : ....... : WriteFileHandle.Flush error: HTTP code 429: "429 Too Many Requests": response body: "{\"logref\":\"c639fc9a-0269-11e7-8580-85f63a4c128e\",\"message\":\"Rate exceeded\",\"code\":\"\"}"

```

I realize these errors are out of rclone's control but something seems fishy with how poorly the mount performs errorwise compared to direct rclone copies to ACDgdrive.  The actual BW usage is great.  Its just the failed partial transfers retries thats the problem.  I'd say 3/4 of my bandwidth is wasted with half uploaded files and then removals/tries sometimes ultimately failing and sometimes succeeding within the 3 retry allowances.  

I can get a debug output if it would help.  This just happened to be what I logged with regular logging for now.

**Comment by calisro** [comment]
To add a little more to this.  I just reran it and I had the same issues but after failing twice it never continued yet stats continued.  I've seen my scripts hanging before but I wasn't entirely sure.  Now I can see it happen in the log.
```

2017/03/06 11:14:25 ERROR : Attempt 1/3 failed with 1 errors and: close /data/Media2/Videos/Series/Homeland/Season.6/Homeland.S06E07.720p.hdtv.h264.mkv: input/output error
2017/03/06 11:14:25 INFO  : .unionfs/Videos/Series/SIX/Season.1/SIX.S01E08.720p.BluRay.x264-DEMAND[ettv]/.actors/Barry_Sloane.jpg_HIDDEN~: Copied (new)
2017/03/06 11:14:27 INFO  : Local file system at /data/Media2: Waiting for checks to finish
2017/03/06 11:14:27 INFO  : Local file system at /data/Media2: Waiting for transfers to finish
2017/03/06 11:15:03 INFO  : 
Transferred:   3.401 GBytes (11.608 MBytes/s)
Errors:                 0
Checks:            169302
Transferred:            8
Elapsed time:        5m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv: 62% done, 13.854 MBytes/s, ETA: 23s

2017/03/06 11:16:08 NOTICE: Videos/Series/Homeland/Season.6/Homeland.S06E07.720p.hdtv.h264.mkv: Removing partially written file on error: close /data/Media2/Videos/Series/Homeland/Season.6/Homeland.S06E07.720p.hdtv.h264.mkv: input/output error
2017/03/06 11:16:08 ERROR : Videos/Series/Homeland/Season.6/Homeland.S06E07.720p.hdtv.h264.mkv: Failed to remove partially written file: remove /data/Media2/Videos/Series/Homeland/Season.6/Homeland.S06E07.720p.hdtv.h264.mkv: no such file or directory
2017/03/06 11:16:08 ERROR : Videos/Series/Homeland/Season.6/Homeland.S06E07.720p.hdtv.h264.mkv: Failed to copy: close /data/Media2/Videos/Series/Homeland/Season.6/Homeland.S06E07.720p.hdtv.h264.mkv: input/output error
2017/03/06 11:16:08 ERROR : Attempt 2/3 failed with 1 errors and: close /data/Media2/Videos/Series/Homeland/Season.6/Homeland.S06E07.720p.hdtv.h264.mkv: input/output error
2017/03/06 11:16:10 INFO  : Local file system at /data/Media2: Waiting for checks to finish
2017/03/06 11:16:10 INFO  : Local file system at /data/Media2: Waiting for transfers to finish
2017/03/06 11:20:03 INFO  : 
Transferred:   3.725 GBytes (6.357 MBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:       10m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 410104h59m9s

2017/03/06 11:25:03 INFO  : 
Transferred:   3.725 GBytes (4.238 MBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:       15m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 0s

2017/03/06 11:30:03 INFO  : 
Transferred:   3.725 GBytes (3.178 MBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:       20m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 0s

2017/03/06 11:35:03 INFO  : 
Transferred:   3.725 GBytes (2.543 MBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:       25m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 0s

2017/03/06 11:40:03 INFO  : 
Transferred:   3.725 GBytes (2.119 MBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:       30m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 0s

2017/03/06 11:45:03 INFO  : 
Transferred:   3.725 GBytes (1.816 MBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:       35m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 0s

2017/03/06 11:50:03 INFO  : 
Transferred:   3.725 GBytes (1.589 MBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:       40m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 0s

2017/03/06 11:55:03 INFO  : 
Transferred:   3.725 GBytes (1.413 MBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:       45m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 0s

2017/03/06 12:00:03 INFO  : 
Transferred:   3.725 GBytes (1.271 MBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:       50m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 0s

2017/03/06 12:05:03 INFO  : 
Transferred:   3.725 GBytes (1.156 MBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:       55m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 0s

2017/03/06 12:10:03 INFO  : 
Transferred:   3.725 GBytes (1.059 MBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:      1h0m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 0s

2017/03/06 12:15:03 INFO  : 
Transferred:   3.725 GBytes (1001.410 kBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:      1h5m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 0s

2017/03/06 12:20:03 INFO  : 
Transferred:   3.725 GBytes (929.881 kBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:     1h10m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 0s

2017/03/06 12:25:03 INFO  : 
Transferred:   3.725 GBytes (867.890 kBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:     1h15m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 0s

2017/03/06 12:30:03 INFO  : 
Transferred:   3.725 GBytes (813.646 kBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:     1h20m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 0s

2017/03/06 12:35:03 INFO  : 
Transferred:   3.725 GBytes (765.784 kBytes/s)
Errors:                 0
Checks:            253957
Transferred:            8
Elapsed time:     1h25m0s
Transferring:
 * ...eason.6/Homeland.S06E07.720p.hdtv.h264.mkv:  0% done, 0 Bytes/s, ETA: 0s

2017/03/06 12:40:03 INFO  : 
Transferred:   3.725 GBytes (723.188 kB
```

**Comment by calisro** [comment]
I'm pretty sure the problem is the mount itself.I actually tried to run another sync on that rclone mount and it seems to be hanging and not doing anything.  Interestingly though, I can browse the mount and even cat a nfo files.  It seems to be 'partially' working.  In order to actually sync, I had to unmount and remount it.  More interesting is when I unmounted it with fusermount, i had to lazy umount (umount -l) to get it to stop.  I then had to manually kill the rclone mount command:
```

└─> ps -ef | grep rclone
robert   22537 28191  0 13:29 pts/30   00:00:00 grep --color=auto rclone
root     29950 29945  1 10:58 pts/30   00:02:27 /usr/sbin/rclone -v --log-file /data/log/rmount-acd.log --config=/home/robert/.rclone.conf mount robacd-cryptp:Media/ /data/Media2 --stats 0 --allow-other --checkers=32 --transfers=32 --no-modtime --timeout 15m --max-read-ahead 1024k --default-permissions --gid 1002 --uid 118 --umask 002 --dir-cache-time 48h
┌[ ✓ robert [/data/bin] $
└─> sudo kill 29950
[sudo] password for robert: 
┌[ ✓ robert [/data/bin] $
└─> /data/bin/cloudfs.sh: line 32: 29950 Terminated              /usr/sbin/rclone -v --log-file /data/log/rmount-acd.log --config="/home/robert/.rclone.conf" mount robacd-cryptp:Media/ $ACD_RCLONE --stats 0 --allow-other --checkers=32 --transfers=32 --no-modtime --timeout 15m --max-read-ahead 1024k --default-permissions --gid $gid --uid $uid --umask $UMASK --dir-cache-time $CACHE

```

**Comment by ncw** [comment]
The main problem is that uploading files via a mount doesn't do retries.  That won't be possible until #711 is in.

That should solve the unreliability of uploads (cross fingers!)

I'd recommend uploading stuff with rclone copy, not via the mount until then.

**Comment by calisro** [comment]
Then i can close this in favor of #711 rather than having yet another ticket.

**Comment by calisro** [comment]
@ncw Shouldn't the lack of I/O be timed out at 5 minutes rather than the hanging?  I realize there are issues with retries but my main issue here is the hang condition.

**Comment by ncw** [comment]
The lack of IO is probably caused by 

    --acd-upload-wait-per-gb duration   Additional time per GB to wait after a failed complete upload to see if it appears. (default 3m0s)

So for your 3.75 GB file rclone waits 11 minutes for it to appear.

**Comment by calisro** [comment]
Yes but I've had it wait for 48 hours with no I/O.  It nevers stops....

**Comment by calisro** [comment]
Even with the failures I would expect it to stop at somepoint.  It seems that when rclone is pointed at a local filesystem (which is actually a rclone mount) it will wait indefinately for the filesystem to return a "okay". It should timeout.

**Comment by ncw** [comment]
The timeouts only apply to network interactions, if you do a local <-> local there will be no timeouts :-(

However the mount should be doing timeouts.  It is possible there is a bug somewhere - can you make me a reproducer?

**Comment by calisro** [comment]
That makes sense.  It reproduces constantly.  I can get you what you need.  Do you need a debug output from the mount?  Anything else? I'll set up whatever you think will help on the mount. It happens for both Amazon and Google Drive so it isn't specific to a cloud type

**Comment by ncw** [comment]
A series of commands to execute to make it happen on my machine would be perfect!

**Comment by calisro** [comment]
The problem is intermittent.  I can't reproduce it on demand but it does reproduce consistently enough.  Lowering the retries helps make it happen more frequently.  I've decreased the checkers also just to make it easy to identify but more transfer threads and it has a higher chance of happening more frequently obviously.

Create a mount:
` rclone -vvv --log-file ~/p.log --checkers 1 --transfers 1 mount robacd-cryptt:/ ~/p --retries 1 --timeout 1m&`

Sync to the mount:
`rclone -vvv --stats 30s --log-file ~/s.log --checkers 1 --transfers 1 copy /data/Media1 ~/p`

Once we do that, we'll see a number copies and at some point we'll see a failure (which is somewhat expected at this point and those aren't really my primary concern with this ticket).  When we see the failure attempts exceed the retries (one in this case but i've increased mine to 10 on my machine), the copy operation will sit there and hang waiting on the mount to error.  It seems that on the mount that copy thread is 'stuck' after a failure.

In the 'stats' in the copy log we'll see at least one file just stop transferring.  Look at 'Transferred:   134.298 MBytes'  like this.  It will never change and will continue indefinitely.  
```

2017/03/16 12:29:55 NOTICE: AudioBooks/1989 - A Time to Kill/Chapter 01.mp3: Removing partially written file on error: close /home/robert/p/AudioBooks/1989 - A Time to Kill/Chapter 01.mp3: input/output error
2017/03/16 12:29:55 ERROR : AudioBooks/1989 - A Time to Kill/Chapter 01.mp3: Failed to remove partially written file: remove /home/robert/p/AudioBooks/1989 - A Time to Kill/Chapter 01.mp3: no such file or directory
2017/03/16 12:29:55 ERROR : AudioBooks/1989 - A Time to Kill/Chapter 01.mp3: Failed to copy: close /home/robert/p/AudioBooks/1989 - A Time to Kill/Chapter 01.mp3: input/output error
2017/03/16 12:30:04 NOTICE: AudioBooks/1989 - A Time to Kill/Chapter 02.mp3: Removing partially written file on error: close /home/robert/p/AudioBooks/1989 - A Time to Kill/Chapter 02.mp3: input/output error
2017/03/16 12:30:04 ERROR : AudioBooks/1989 - A Time to Kill/Chapter 02.mp3: Failed to remove partially written file: remove /home/robert/p/AudioBooks/1989 - A Time to Kill/Chapter 02.mp3: no such file or directory
2017/03/16 12:30:04 ERROR : AudioBooks/1989 - A Time to Kill/Chapter 02.mp3: Failed to copy: close /home/robert/p/AudioBooks/1989 - A Time to Kill/Chapter 02.mp3: input/output error
2017/03/16 12:30:08 INFO  : AudioBooks/1989 - A Time to Kill/Chapter 03.mp3: Copied (new)
2017/03/16 12:30:11 INFO  : AudioBooks/1989 - A Time to Kill/Chapter 04.mp3: Copied (new)
2017/03/16 12:30:12 INFO  : AudioBooks/1989 - A Time to Kill/Chapter 05.mp3: Copied (new)
2017/03/16 12:30:13 INFO  : AudioBooks/The Subtle Art of Not Giving a F ck by Mark Manson.m4b: Updated modification time in destination
2017/03/16 12:30:13 DEBUG : AudioBooks/The Subtle Art of Not Giving a F ck by Mark Manson.m4b: Unchanged skipping
2017/03/16 12:30:13 DEBUG : AudioBooks/find her.mp3: Modification times differ by 9469h52m27.916930185s: 2016-02-15 21:35:28.507069815 -0500 EST, 2017-03-16 12:27:56.424 -0400 EDT
2017/03/16 12:30:14 INFO  : AudioBooks/1989 - A Time to Kill/Chapter 06.mp3: Copied (new)
2017/03/16 12:30:17 INFO  : AudioBooks/1989 - A Time to Kill/Chapter 07.mp3: Copied (new)
2017/03/16 12:30:18 INFO  : AudioBooks/1989 - A Time to Kill/Chapter 08.mp3: Copied (new)
2017/03/16 12:30:20 INFO  : AudioBooks/1989 - A Time to Kill/Chapter 09.mp3: Copied (new)
2017/03/16 12:30:21 INFO  : 
Transferred:   134.298 MBytes (1.119 MBytes/s)
Errors:                 4
Checks:                 3
Transferred:            8
Elapsed time:        2m0s
Checking:
 * AudioBooks/find her.mp3
Transferring:
 * ...Books/1989 - A Time to Kill/Chapter 10.mp3: 29% done, 3.902 MBytes/s, ETA: 2s

2017/03/16 12:30:51 INFO  : 
Transferred:   134.298 MBytes (916.788 kBytes/s)
Errors:                 4
Checks:                 3
Transferred:            8
Elapsed time:       2m30s
Checking:
 * AudioBooks/find her.mp3
Transferring:
 * ...Books/1989 - A Time to Kill/Chapter 10.mp3: 29% done, 540.338 kBytes/s, ETA: 18s

2017/03/16 12:31:21 INFO  : 
Transferred:   134.298 MBytes (763.993 kBytes/s)
Errors:                 4
Checks:                 3
Transferred:            8
Elapsed time:        3m0s
Checking:
 * AudioBooks/find her.mp3
Transferring:
 * ...Books/1989 - A Time to Kill/Chapter 10.mp3: 29% done, 73.072 kBytes/s, ETA: 2m13s

2017/03/16 12:31:24 INFO  : AudioBooks/find her.mp3: Updated modification time in destination
2017/03/16 12:31:24 DEBUG : AudioBooks/find her.mp3: Unchanged skipping
2017/03/16 12:31:51 INFO  : 
Transferred:   134.298 MBytes (654.854 kBytes/s)
Errors:                 4
Checks:                 4
Transferred:            8
Elapsed time:       3m30s
Transferring:
 * ...Books/1989 - A Time to Kill/Chapter 10.mp3: 29% done, 9.882 kBytes/s, ETA: 16m27s

2017/03/16 12:32:21 INFO  : 
Transferred:   134.298 MBytes (572.998 kBytes/s)
Errors:                 4
Checks:                 4
Transferred:            8
Elapsed time:        4m0s
Transferring:
 * ...Books/1989 - A Time to Kill/Chapter 10.mp3: 29% done, 1.336 kBytes/s, ETA: 2h1m45s

2017/03/16 12:32:51 INFO  : 
Transferred:   134.298 MBytes (509.332 kBytes/s)
Errors:                 4
Checks:                 4
Transferred:            8
Elapsed time:       4m30s
Transferring:
 * ...Books/1989 - A Time to Kill/Chapter 10.mp3: 29% done, 185 Bytes/s, ETA: 15h0m21s

2017/03/16 12:33:21 INFO  : 
Transferred:   134.298 MBytes (458.399 kBytes/s)
Errors:                 4
Checks:                 4
Transferred:            8
Elapsed time:        5m0s
Transferring:
 * ...Books/1989 - A Time to Kill/Chapter 10.mp3: 29% done, 25 Bytes/s, ETA: 110h57m44s

2017/03/16 12:33:51 INFO  : 
Transferred:   134.298 MBytes (416.728 kBytes/s)
Errors:                 4
Checks:                 4
Transferred:            8
Elapsed time:       5m30s
Transferring:
 * ...Books/1989 - A Time to Kill/Chapter 10.mp3: 29% done, 3 Bytes/s, ETA: 820h30m56s

```

Once that happens, I can't even terminate the rclone copy with a 'ctrl-c'.  I must kill it "-9" and then it is left defunct.

**Comment by djsecrist** [comment]
I am having the same issue, I believe. I am using nzbget to download files to my local disk and then extract the results to Google Drive (encrypted). It will work beautifully for some period of time, but eventually it will always get hung up in the same manner as described in the previous messages. My quick fix is to just reboot the server, but manually babysitting the process isn't a sustainable workaround.

**Comment by calisro** [comment]
Yes, mine is consistent in the failures.  I've  increased 'retries' to 20 and that seems to help in that it has a higher chance of not getting hung up after the retries exceed.  Even right now I can see the hang.  Other threads process fine but slowly they all hang up.

```

2017/03/29 08:54:22 INFO  : 
Transferred:   1.414 GBytes (4.825 MBytes/s)
Errors:                 9
Checks:              5274
Transferred:           58
Elapsed time:        5m0s
Checking:
 * Videos/Movies/Aladdin.(1992).720P.BRRip.x264/Aladdin.(1992).720P.BRRip.x264.nfo
Transferring:
 *                    Pictures/2017/DSC_0121.JPG:  4% done, 0 Bytes/s, ETA: 170479h45m23s
 *                    Pictures/2017/DSC_0122.JPG:  7% done, 0 Bytes/s, ETA: 43852h20m37s
 * ...ason.2/Blindspot.S02E16.720p.hdtv.h264.mkv: 77% done, 2.499 MBytes/s, ETA: 1m16s
 * ...tury.Women.(2016).720p.bluray.h264.aac.mp4: 79% done, 2.507 MBytes/s, ETA: 1m16s

2017/03/29 08:55:43 ERROR : Videos/Series/Blindspot/Season.2/Blindspot.S02E16.720p.hdtv.h264.mkv: Failed to copy: chtimes /data/Media2/Videos/Series/Blindspot/Season.2/Blindspot.S02E16.720p.hdtv.h264.mkv: no such file or directory
2017/03/29 08:55:45 INFO  : Videos/Movies/20th.Century.Women.(2016).720p.bluray.h264.aac/20th.Century.Women.(2016).720p.bluray.h264.aac.mp4: Copied (new)
2017/03/29 08:55:47 INFO  : .unionfs/Videos/Series/Game.of.Thrones/Season.6/Game.of.Thrones.S06E03.1080p.HDTV.x264-BATV[ettv]-thumb.jpg_HIDDEN~: Copied (new)
2017/03/29 08:55:48 INFO  : Videos/Movies/20th.Century.Women.(2016).720p.bluray.h264.aac/.actors/Annette_Bening.jpg: Copied (new)
2017/03/29 08:59:22 INFO  : 
Transferred:   2.859 GBytes (4.879 MBytes/s)
Errors:                10
Checks:              5275
Transferred:           61
Elapsed time:       10m0s
Transferring:
 *                    Pictures/2017/DSC_0121.JPG:  4% done, 0 Bytes/s, ETA: 1217378h48m37.349638144s
 *                    Pictures/2017/DSC_0122.JPG:  7% done, 0 Bytes/s, ETA: 1707882h59m7.715026944s
 * ...le.Lies.S01E05.1080p.webrip.h264.dd5.1.mkv:  9% done, 2.501 MBytes/s, ETA: 34m7s
 * ...on.5/Bates.Motel.S05E05.720p.hdtv.h264.mkv: 67% done, 2.494 MBytes/s, ETA: 1m51s

2017/03/29 09:01:35 INFO  : Videos/Series/Bates.Motel/Season.5/Bates.Motel.S05E05.720p.hdtv.h264.mkv: Copied (new)
2017/03/29 09:04:22 INFO  : 
Transferred:   4.324 GBytes (4.919 MBytes/s)
Errors:                10
Checks:              5275
Transferred:           62
Elapsed time:       15m0s
Transferring:
 *                    Pictures/2017/DSC_0121.JPG:  4% done, 0 Bytes/s, ETA: 0s
 *                    Pictures/2017/DSC_0122.JPG:  7% done, 0 Bytes/s, ETA: 0s
 * ...eason.2/Billions.S02E05.720p.hdtv.h264.mkv: 48% done, 2.506 MBytes/s, ETA: 2m59s
 * ...le.Lies.S01E05.1080p.webrip.h264.dd5.1.mkv: 23% done, 2.499 MBytes/s, ETA: 28m45s

```

These two lines will never end and never abort.  The only thing that can be done is kill -9 on the sync and the calling shell script. 
 *                    Pictures/2017/DSC_0121.JPG:  4% done, 0 Bytes/s, ETA: 0s
 *                    Pictures/2017/DSC_0122.JPG:  7% done, 0 Bytes/s, ETA: 0s

@ncw Is there anything I can get you further to help?  The issue is pretty consistent and hte test case should be able to demonstrate it but im happy to provide anything to help you.

**Comment by mrmachine** [comment]
I am also seeing the same thing, except that it is Sick Rage that I can't terminate (instead of an `rclone copy` command). If I unmount the remote, then Sick Rage becomes responsive again.

https://forum.rclone.org/t/sick-rage-stalls-during-move-to-rclone-mount-gsuite/1703

**Comment by ncw** [comment]
@calisro Can you still replicate the stuck transfers with [the latest beta](https://beta.rclone.org)?

If so I'd like to see logs with `-vv` of just the file that gets stuck, so if you could grep your logs for one which does get stuck that would be very useful.  I want to find out what has happened to it!

Also if you do get one which gets stuck, can you `kill -QUIT` or press CTRL-\ on the `rclone copy` process.  This will kill it, but generate a backtrace of every goroutine (which might be quite large!).  That should show where the process has got stuck.

**Comment by calisro** [comment]
I'll work on that and re-verify that this is still a problem with the latest.  When it was stuck before,  a kill -9 would terminate it and leave it defunct so I'm not sure if kill -QUIT will work but I'll give it a try if I can replicate it by turning down the retries again.

**Comment by calisro** [comment]
I've been unable to recreate the hangs.  Somewhere between  v1.35-162-gb2a4ea9β and the latest beta, the mount has become more stable.  I'll close this for now and re-open if I find it re-occurs.
